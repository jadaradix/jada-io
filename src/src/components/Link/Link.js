import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components'

const StyledLink = styled.a`
  display: block;
  padding-bottom: 2px;
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", sans-serif;
  font-weight: 600;
  font-size: 1.2rem;
  letter-spacing: 0.1px;
  color: rgb(255, 45, 85);
`;

const Link = ({ children, href }) => (
  <StyledLink href={href}>
    {children}
  </StyledLink>
);

Link.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]),
  href: PropTypes.string.isRequired
};

Link.defaultProps = {
};

export default Link;
