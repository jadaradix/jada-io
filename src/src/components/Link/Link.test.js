import React from 'react';
import renderer from 'react-test-renderer';
import Link from './Link';

it('renders', () => {
  const tree = renderer
    .create(<Link href='https://github.com'>GitHub</Link>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
