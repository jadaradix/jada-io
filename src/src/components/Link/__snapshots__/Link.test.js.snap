// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
.c0 {
  display: block;
  padding-bottom: 2px;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",sans-serif;
  font-weight: 600;
  font-size: 1.2rem;
  -webkit-letter-spacing: 0.1px;
  -moz-letter-spacing: 0.1px;
  -ms-letter-spacing: 0.1px;
  letter-spacing: 0.1px;
  color: rgb(255,45,85);
}

<a
  className="c0"
  href="https://github.com"
>
  GitHub
</a>
`;
